platform/default/mbgl/util/run_loop_scheduled_task.cpp

```diff
@@ -0,0 +1,59 @@
+#include <mbgl/util/run_loop_scheduled_task.hpp>
+
+#include <functional>
+
+namespace  {
+
+using namespace mbgl;
+
+// Wraps a functor in a movable/copyable type
+// by moving it to the heap in a shared_ptr
+template <class Fn>
+class FunctorWrapper
+{
+public:
+    FunctorWrapper(Fn fn_) : fn(std::make_shared<Fn>(std::move(fn_))) {};
+    void operator()(){
+        (*fn)();
+    }
+private:
+    std::shared_ptr<Fn> fn;
+};
```

Comment:
We have the combination of `WorkRequest`/`WorkTask` which is very similar to this. Can this be reused here?